### PR TITLE
[Do Not Merge] RecoveryServices.Backup update for latest release

### PR DESCRIPTION
**Pull request by sumitmal**
## Description
https://github.com/Azure/azure-rest-api-specs/pull/3395
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/MabJob.cs [hunk 1]
```diff
@@ -106,11 +105,10 @@ public MabJob(string entityFriendlyName = default(string), string backupManageme
         /// 'DPMContainer', 'AzureBackupServerContainer', 'MABContainer',
         /// 'Cluster', 'AzureSqlContainer', 'Windows', 'VCenter',
         /// 'VMAppContainer', 'SQLAGWorkLoadContainer', 'StorageContainer',
-        /// 'GenericContainer', 'SqlCluster', 'ExchangeDAG', 'SharepointFarm',
-        /// 'HyperVCluster', 'WindowsClient'
+        /// 'GenericContainer'
         /// </summary>
         [JsonProperty(PropertyName = "mabServerType")]
-        public string MabServerType { get; set; }
```
Comment (dsgouda) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/MabJob.cs]: These are breaking changes, please bump the assembly version to `3.1.0-preview`

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Recovery Services Backup Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Recovery Services Backup Resources.")]
 
-[assembly: AssemblyVersion("3.0.0.0")]
-[assembly: AssemblyFileVersion("3.0.0.0")]
+[assembly: AssemblyVersion("3.0.0.1")]
```
Comment (dsgouda) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs]: `AssemblyVersion` remains the same, `AssemblyFileVersion` must be updated

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup.csproj [hunk 3]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.RecoveryServices.Backup</PackageId>
 		<Description>Provides developers with libraries for the updated recovery services backup feature under Azure Resource manager.</Description>
-		<Version>3.0.0-preview</Version>
+		<Version>3.0.1-preview</Version>
 		<AssemblyName>Microsoft.Azure.Management.RecoveryServices.Backup</AssemblyName>
 		<PackageTags>Recoveryservices;Backup;</PackageTags>
 		<PackageReleaseNotes/>
```
Comment (dsgouda) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup.csproj]: Please add meaningful information in the PackageReleaseNotes similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj#L13)

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please regenerate the code using the generate.ps1 script [here](https://github.com/sumitmal/azure-sdk-for-net/blob/2c8e0d356109141ad109f6688824abfaf538753f/src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/generate.ps1)
 and commit all changes.

**Comment by dsgouda** [comment]
@sumitmal Please join the Azure org on github [here](https://aka.ms/azuregithub) to trigger the CI builds

**Comment by dsgouda** [comment]
@sumitmal Please join the Azure org as suggested. Will close the PR by next week if there is no activity

**Comment by sumitmal** [comment]
Resolved following comments @dsgouda 
1. Bumped the minor version
2. Generated.ps1 used to generate
3. Update package release notes.
4. Joined Azure.

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Recovery Services Backup Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Recovery Services Backup Resources.")]
 
-[assembly: AssemblyVersion("3.0.0.0")]
-[assembly: AssemblyFileVersion("3.0.0.0")]
+[assembly: AssemblyVersion("3.1.0.0")]
```
Comment (dsgouda) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs]: Please keep the `AssemblyVersion` unchanged here

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from the one comment

**Comment by sumitmal** [comment]
@dsgouda  done

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge on CIs passing

**Review by shahabhijeet — APPROVED** [review]
Ignore the fact that this was a preview package.
Looks good.

**Pull request merged**